core/templates/dev/head/domain/topic/TopicObjectFactory.js

```diff
@@ -132,13 +136,21 @@ oppia.factory('TopicObjectFactory', ['SubtopicObjectFactory',
             }
           }
           this._subtopics.splice(i, 1);
-          i--;
           subtopicDeleted = true;
+          break;
         }
       }
       if (isNewlyCreated && subtopicDeleted) {
+        for (var i = 0; i < this._subtopics.length; i++) {
+          if (this._subtopics[i].getId() > subtopicId) {
+            this._subtopics[i].decrementId();
+          }
+        }
         this._nextSubtopicId--;
       }
+      if (!subtopicDeleted) {
```

Comment:
Move this to line 143 so that you break early if there's an error. Then, in the if clause that's currently on line 143, just make it "if (isNewlyCreated)".

Reply:
Done.